packages/vscode-extension/lib/inspector.js

```diff
@@ -0,0 +1,112 @@
+const { Dimensions, findNodeHandle } = require("react-native");
+const getInspectorDataForViewAtPoint = require("react-native/Libraries/Inspector/getInspectorDataForViewAtPoint");
+
+const getInspectorDataForInstance = (node) => {
+  const renderers = Array.from(window.__REACT_DEVTOOLS_GLOBAL_HOOK__?.renderers?.values());
+  if (!renderers) {
+    return {};
+  }
+  for (const renderer of renderers) {
+    if (renderer.rendererConfig?.getInspectorDataForInstance) {
+      const data = renderer.rendererConfig.getInspectorDataForInstance(node);
+      return data ?? {};
+    }
+  }
+  return {};
+};
+
+const createStackElement = (
+  frame, name, source
+) => (
+  {
+    componentName: name,
+    source: {
+      fileName: source.fileName,
+      line0Based: source.lineNumber - 1,
+      column0Based: source.columnNumber - 1,
+    },
+    frame,
+  });
+
+const traverseComponentsTreeUp = (node, stack) => {
+  // Optimization: we break after reaching fiber node corresponding to OffscreenComponent (with tag 22).
+  // https://github.com/facebook/react/blob/c3570b158d087eb4e3ee5748c4bd9360045c8a26/packages/react-reconciler/src/ReactWorkTags.js#L62
+  if (!node || node.tag === 22) {
+    return stack;
+  } else {
+    const data = getInspectorDataForInstance(node);
+    const item = data.hierarchy[data.hierarchy.length - 1];
```

Comment:
getInspectorDataForInstance can return empty objects in which case this line will crash